go panic on `conduit stat ns`

On master, when I run `conduit stat ns`, I get a `panic: runtime error: slice bounds out of range` error.

```
 (master) $ go run cli/main.go stat ns --kubeconfig ~/.kube/config --api-addr localhost:8085
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/runconduit/conduit/cli/cmd.renderStats(0xc420076580, 0x1de0155, 0xa, 0xc42003b490, 0x0, 0x0)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/cli/cmd/stat.go:146 +0x217
github.com/runconduit/conduit/cli/cmd.requestStatsFromAPI(0x1efa760, 0xc420076560, 0xc420346ea0, 0xc42003b490, 0xc420346ea0, 0x0, 0x0, 0x1e681e0)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/cli/cmd/stat.go:136 +0x20b
github.com/runconduit/conduit/cli/cmd.newCmdStat.func1(0xc42035afc0, 0xc420348b90, 0x1, 0x5, 0x0, 0x0)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/cli/cmd/stat.go:105 +0x1d0
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).execute(0xc42035afc0, 0xc420348b40, 0x5, 0x5, 0xc42035afc0, 0xc420348b40)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:698 +0x46d
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x264b680, 0x0, 0xc4201e7f68, 0x1b60eba)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/runconduit/conduit/vendor/github.com/spf13/cobra.(*Command).Execute(0x264b680, 0xc4201e7f78, 0xc4200ba058)
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/Users/mars/workspace/go/src/github.com/runconduit/conduit/cli/main.go:10 +0x2d
exit status 2
```

--- a/cli/cmd/stat.go
+++ b/cli/cmd/stat.go
@@ -223,15 +223,15 @@ func writeStatsToBuffer(resp *pb.StatSummaryResponse, reqResourceType string, w 
 		os.Exit(0)
 	}
 
-	lastDisplayedStat := true // don't print a newline after the final stat
 	switch reqResourceType {
 	case k8s.All:
+		firstDisplayedStat := true // don't print a newline before the first stat
 		for _, resourceType := range k8s.StatAllResourceTypes {
 			if stats, ok := statTables[resourceType]; ok {
-				if !lastDisplayedStat {
+				if !firstDisplayedStat {
 					fmt.Fprint(w, "\n")
 				}
-				lastDisplayedStat = false
+				firstDisplayedStat = false
 				printStatTable(stats, resourceType, w, maxNameLength, maxNamespaceLength, options)
 			}
 		}

--- a/cli/cmd/stat.go
+++ b/cli/cmd/stat.go
@@ -224,18 +224,20 @@ func writeStatsToBuffer(resp *pb.StatSummaryResponse, reqResourceType string, w 
 	}
 
 	lastDisplayedStat := true // don't print a newline after the final stat
-
-	for _, resourceType := range k8s.StatAllResourceTypes {
-		if stats, ok := statTables[resourceType]; ok {
-			if !lastDisplayedStat {
-				fmt.Fprint(w, "\n")
-			}
-			lastDisplayedStat = false
-			if reqResourceType == k8s.All {
+	switch reqResourceType {
+	case k8s.All:
+		for _, resourceType := range k8s.StatAllResourceTypes {
+			if stats, ok := statTables[resourceType]; ok {
+				if !lastDisplayedStat {
+					fmt.Fprint(w, "\n")
+				}
+				lastDisplayedStat = false
 				printStatTable(stats, resourceType, w, maxNameLength, maxNamespaceLength, options)
-			} else {
-				printStatTable(stats, "", w, maxNameLength, maxNamespaceLength, options)
 			}
+		}
+	default:
+		if stats, ok := statTables[reqResourceType]; ok {
+			printStatTable(stats, "", w, maxNameLength, maxNamespaceLength, options)
 		}
 	}
 }

--- a/cli/cmd/stat_test.go
+++ b/cli/cmd/stat_test.go
@@ -31,7 +31,7 @@ emoji      1/2   100.00%   2.0rps         123ms         123ms         123ms     
 		}
 
 		if output != expectedOutput {
-			t.Fatalf("Wrong output:\n expected: \n%v\n, got: \n%v", expectedOutput, output)
+			t.Fatalf("Wrong output:\n expected: \n%s\n, got: \n%s", expectedOutput, output)
 		}
 	})
 }

--- a/cli/cmd/stat_test.go
+++ b/cli/cmd/stat_test.go
@@ -1,0 +1,37 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/runconduit/conduit/controller/api/public"
+)
+
+func TestStat(t *testing.T) {
+	t.Run("Returns namespace stats", func(t *testing.T) {
+		mockClient := &public.MockConduitApiClient{}
+
+		response := public.GenStatSummaryResponse("emoji", "namespaces", "emojivoto", 1, 2, 0)
+
+		mockClient.StatSummaryResponseToReturn = &response
+
+		expectedOutput := `NAME    MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   SECURED
+emoji      1/2   100.00%   2.0rps         123ms         123ms         123ms      100%
+`
+
+		options := newStatOptions()
+		args := []string{"ns"}
+		req, err := buildStatSummaryRequest(args, options)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		output, err := requestStatsFromAPI(mockClient, req, options)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		if output != expectedOutput {
+			t.Fatalf("Wrong output:\n expected: \n%v\n, got: \n%v", expectedOutput, output)
+		}
+	})
+}

--- a/controller/api/public/stat_summary_test.go
+++ b/controller/api/public/stat_summary_test.go
@@ -43,44 +43,6 @@ func genPromSample(resName string, resType string, resNs string, classification 
 	}
 }
 
-func genStatSummaryResponse(resName, resType, resNs string, meshedPods uint64, runningPods uint64, failedPods uint64) pb.StatSummaryResponse {
-	return pb.StatSummaryResponse{
-		Response: &pb.StatSummaryResponse_Ok_{ // https://github.com/golang/protobuf/issues/205
-			Ok: &pb.StatSummaryResponse_Ok{
-				StatTables: []*pb.StatTable{
-					&pb.StatTable{
-						Table: &pb.StatTable_PodGroup_{
-							PodGroup: &pb.StatTable_PodGroup{
-								Rows: []*pb.StatTable_PodGroup_Row{
-									&pb.StatTable_PodGroup_Row{
-										Resource: &pb.Resource{
-											Namespace: resNs,
-											Type:      resType,
-											Name:      resName,
-										},
-										Stats: &pb.BasicStats{
-											SuccessCount:    123,
-											FailureCount:    0,
-											LatencyMsP50:    123,
-											LatencyMsP95:    123,
-											LatencyMsP99:    123,
-											TlsRequestCount: 123,
-										},
-										TimeWindow:      "1m",
-										MeshedPodCount:  meshedPods,
-										RunningPodCount: runningPods,
-										FailedPodCount:  failedPods,
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-}
-
 func genEmptyResponse() pb.StatSummaryResponse {
 	return pb.StatSummaryResponse{
 		Response: &pb.StatSummaryResponse_Ok_{ // https://github.com/golang/protobuf/issues/205
@@ -229,7 +191,7 @@ status:
 					},
 					TimeWindow: "1m",
 				},
-				expectedResponse: genStatSummaryResponse("emoji", "deployments", "emojivoto", 1, 2, 0),
+				expectedResponse: GenStatSummaryResponse("emoji", "deployments", "emojivoto", 1, 2, 0),
 			},
 		}
 
@@ -271,7 +233,7 @@ status:
 					`histogram_quantile(0.99, sum(irate(response_latency_ms_bucket{direction="inbound", namespace="emojivoto", pod="emojivoto-1"}[1m])) by (le, namespace, pod))`,
 					`sum(increase(response_total{direction="inbound", namespace="emojivoto", pod="emojivoto-1"}[1m])) by (namespace, pod, classification, tls)`,
 				},
-				expectedResponse: genStatSummaryResponse("emojivoto-1", "pods", "emojivoto", 1, 1, 0),
+				expectedResponse: GenStatSummaryResponse("emojivoto-1", "pods", "emojivoto", 1, 1, 0),
 			},
 		}
 
@@ -763,7 +725,7 @@ status:
 						},
 						TimeWindow: "1m",
 					},
-					expectedResponse: genStatSummaryResponse("emoji", "deployments", "emojivoto", 1, 2, 1),
+					expectedResponse: GenStatSummaryResponse("emoji", "deployments", "emojivoto", 1, 2, 1),
 				},
 			}
 

--- a/controller/api/public/test_helper.go
+++ b/controller/api/public/test_helper.go
@@ -125,3 +125,41 @@ func (m *MockProm) LabelValues(ctx context.Context, label string) (model.LabelVa
 func (m *MockProm) Series(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) ([]model.LabelSet, error) {
 	return nil, nil
 }
+
+func GenStatSummaryResponse(resName, resType, resNs string, meshedPods uint64, runningPods uint64, failedPods uint64) pb.StatSummaryResponse {
+	return pb.StatSummaryResponse{
+		Response: &pb.StatSummaryResponse_Ok_{ // https://github.com/golang/protobuf/issues/205
+			Ok: &pb.StatSummaryResponse_Ok{
+				StatTables: []*pb.StatTable{
+					&pb.StatTable{
+						Table: &pb.StatTable_PodGroup_{
+							PodGroup: &pb.StatTable_PodGroup{
+								Rows: []*pb.StatTable_PodGroup_Row{
+									&pb.StatTable_PodGroup_Row{
+										Resource: &pb.Resource{
+											Namespace: resNs,
+											Type:      resType,
+											Name:      resName,
+										},
+										Stats: &pb.BasicStats{
+											SuccessCount:    123,
+											FailureCount:    0,
+											LatencyMsP50:    123,
+											LatencyMsP95:    123,
+											LatencyMsP99:    123,
+											TlsRequestCount: 123,
+										},
+										TimeWindow:      "1m",
+										MeshedPodCount:  meshedPods,
+										RunningPodCount: runningPods,
+										FailedPodCount:  failedPods,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}

--- a/test/stat/stat_test.go
+++ b/test/stat/stat_test.go
@@ -82,11 +82,7 @@ func TestCliStatForConduitNamespace(t *testing.T) {
 				return err
 			}
 
-			if err := validateRowStats(TestHelper.GetConduitNamespace(), "4/4", rowStats); err != nil {
-				return err
-			}
-
-			return nil
+			return validateRowStats(TestHelper.GetConduitNamespace(), "4/4", rowStats)
 		})
 		if err != nil {
 			t.Fatal(err.Error())

--- a/test/stat/stat_test.go
+++ b/test/stat/stat_test.go
@@ -45,78 +45,61 @@ var controllerDeployments = []string{"controller", "grafana", "prometheus", "web
 // failed attempts will eventually be recorded in the stats that we're
 // requesting, and the test will pass.
 func TestCliStatForConduitNamespace(t *testing.T) {
+	t.Run("test conduit stat deploy", func(t *testing.T) {
+		err := TestHelper.RetryFor(20*time.Second, func() error {
+			out, err := TestHelper.ConduitRun("stat", "deploy", "-n", TestHelper.GetConduitNamespace())
+			if err != nil {
+				t.Fatalf("Unexpected stat error: %v", err)
+			}
 
-	// test `conduit stat deploy -n <namespace>`
-	err := TestHelper.RetryFor(20*time.Second, func() error {
-		out, err := TestHelper.ConduitRun("stat", "deploy", "-n", TestHelper.GetConduitNamespace())
-		if err != nil {
-			t.Fatalf("Unexpected stat error: %v", err)
-		}
-
-		rowStats, err := parseRows(out, 4, false)
-		if err != nil {
-			return err
-		}
-
-		for _, name := range controllerDeployments {
-			if err := validateRowStats(name, rowStats); err != nil {
+			rowStats, err := parseRows(out, 4)
+			if err != nil {
 				return err
 			}
-		}
 
-		return nil
-	})
-	if err != nil {
-		t.Fatal(err.Error())
-	}
-
-	// test `conduit stat ns`
-	err = TestHelper.RetryFor(20*time.Second, func() error {
-		out, err := TestHelper.ConduitRun("stat", "ns")
-		if err != nil {
-			t.Fatalf("Unexpected stat error: %v", err)
-		}
-
-		rowStats, err := parseRows(out, 2, true)
-		if err != nil {
-			return err
-		}
-
-		conduitNsFound := false
-		emojivotoNsFound := false
-		for _, row := range rowStats {
-			if row.name == "emojivoto" {
-				emojivotoNsFound = true
+			for _, name := range controllerDeployments {
+				if err := validateRowStats(name, "1/1", rowStats); err != nil {
+					return err
+				}
 			}
-			if row.name == TestHelper.GetConduitNamespace() {
-				conduitNsFound = true
-			}
-		}
-		if conduitNsFound == false {
-			t.Fatalf("Expected conduit stat to find stats for namespace: %s", TestHelper.GetConduitNamespace())
-		}
-		if emojivotoNsFound == false {
-			t.Fatalf("Expected conduit stat to find stats for namespace: emojivoto")
-		}
 
-		return nil
+			return nil
+		})
+		if err != nil {
+			t.Fatal(err.Error())
+		}
 	})
-	if err != nil {
-		t.Fatal(err.Error())
-	}
+
+	t.Run("test conduit stat namespace", func(t *testing.T) {
+		err := TestHelper.RetryFor(20*time.Second, func() error {
+			out, err := TestHelper.ConduitRun("stat", "ns", TestHelper.GetConduitNamespace())
+			if err != nil {
+				t.Fatalf("Unexpected stat error: %v", err)
+			}
+
+			rowStats, err := parseRows(out, 1)
+			if err != nil {
+				return err
+			}
+
+			if err := validateRowStats(TestHelper.GetConduitNamespace(), "4/4", rowStats); err != nil {
+				return err
+			}
+
+			return nil
+		})
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+	})
 }
 
 // check that expectedRowCount rows have been returned
-// if checkMin is set, check that a minimum of expectedRowCount rows have been returned
-func checkRowCount(out string, expectedRowCount int, checkMin bool) ([]string, error) {
+func checkRowCount(out string, expectedRowCount int) ([]string, error) {
 	rows := strings.Split(out, "\n")
 	rows = rows[1 : len(rows)-1] // strip header and trailing newline
 
-	if checkMin && len(rows) < expectedRowCount {
-		return nil, fmt.Errorf(
-			"Expected at least [%d] rows in stat output, got [%d]; full output:\n%s",
-			expectedRowCount, len(rows), strings.Join(rows, "\n"))
-	} else if !checkMin && len(rows) != expectedRowCount {
+	if len(rows) != expectedRowCount {
 		return nil, fmt.Errorf(
 			"Expected [%d] rows in stat output, got [%d]; full output:\n%s",
 			expectedRowCount, len(rows), strings.Join(rows, "\n"))
@@ -125,8 +108,8 @@ func checkRowCount(out string, expectedRowCount int, checkMin bool) ([]string, e
 	return rows, nil
 }
 
-func parseRows(out string, expectedRowCount int, checkMin bool) (map[string]*rowStat, error) {
-	rows, err := checkRowCount(out, expectedRowCount, checkMin)
+func parseRows(out string, expectedRowCount int) (map[string]*rowStat, error) {
+	rows, err := checkRowCount(out, expectedRowCount)
 	if err != nil {
 		return nil, err
 	}
@@ -157,13 +140,12 @@ func parseRows(out string, expectedRowCount int, checkMin bool) (map[string]*row
 	return rowStats, nil
 }
 
-func validateRowStats(name string, rowStats map[string]*rowStat) error {
+func validateRowStats(name, expectedMeshCount string, rowStats map[string]*rowStat) error {
 	stat, ok := rowStats[name]
 	if !ok {
 		return fmt.Errorf("No stats found for [%s]", name)
 	}
 
-	expectedMeshCount := "1/1"
 	if stat.meshed != expectedMeshCount {
 		return fmt.Errorf("Expected mesh count [%s] for [%s], got [%s]",
 			expectedMeshCount, name, stat.meshed)

--- a/test/stat/stat_test.go
+++ b/test/stat/stat_test.go
@@ -46,13 +46,14 @@ var controllerDeployments = []string{"controller", "grafana", "prometheus", "web
 // requesting, and the test will pass.
 func TestCliStatForConduitNamespace(t *testing.T) {
 
+	// test `conduit stat deploy -n <namespace>`
 	err := TestHelper.RetryFor(20*time.Second, func() error {
 		out, err := TestHelper.ConduitRun("stat", "deploy", "-n", TestHelper.GetConduitNamespace())
 		if err != nil {
 			t.Fatalf("Unexpected stat error: %v", err)
 		}
 
-		rowStats, err := parseRows(out)
+		rowStats, err := parseRows(out, 4, false)
 		if err != nil {
 			return err
 		}
@@ -69,17 +70,65 @@ func TestCliStatForConduitNamespace(t *testing.T) {
 		t.Fatal(err.Error())
 	}
 
+	// test `conduit stat ns`
+	err = TestHelper.RetryFor(20*time.Second, func() error {
+		out, err := TestHelper.ConduitRun("stat", "ns")
+		if err != nil {
+			t.Fatalf("Unexpected stat error: %v", err)
+		}
+
+		rowStats, err := parseRows(out, 2, true)
+		if err != nil {
+			return err
+		}
+
+		conduitNsFound := false
+		emojivotoNsFound := false
+		for _, row := range rowStats {
+			if row.name == "emojivoto" {
+				emojivotoNsFound = true
+			}
+			if row.name == TestHelper.GetConduitNamespace() {
+				conduitNsFound = true
+			}
+		}
+		if conduitNsFound == false {
+			t.Fatalf("Expected conduit stat to find stats for namespace: %s", TestHelper.GetConduitNamespace())
+		}
+		if emojivotoNsFound == false {
+			t.Fatalf("Expected conduit stat to find stats for namespace: emojivoto")
+		}
+
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
 }
 
-func parseRows(out string) (map[string]*rowStat, error) {
+// check that expectedRowCount rows have been returned
+// if checkMin is set, check that a minimum of expectedRowCount rows have been returned
+func checkRowCount(out string, expectedRowCount int, checkMin bool) ([]string, error) {
 	rows := strings.Split(out, "\n")
 	rows = rows[1 : len(rows)-1] // strip header and trailing newline
 
-	expectedRowCount := 4
-	if len(rows) != expectedRowCount {
+	if checkMin && len(rows) < expectedRowCount {
+		return nil, fmt.Errorf(
+			"Expected at least [%d] rows in stat output, got [%d]; full output:\n%s",
+			expectedRowCount, len(rows), strings.Join(rows, "\n"))
+	} else if !checkMin && len(rows) != expectedRowCount {
 		return nil, fmt.Errorf(
 			"Expected [%d] rows in stat output, got [%d]; full output:\n%s",
 			expectedRowCount, len(rows), strings.Join(rows, "\n"))
+	}
+
+	return rows, nil
+}
+
+func parseRows(out string, expectedRowCount int, checkMin bool) (map[string]*rowStat, error) {
+	rows, err := checkRowCount(out, expectedRowCount, checkMin)
+	if err != nil {
+		return nil, err
 	}
 
 	rowStats := make(map[string]*rowStat)

